packages/perspective-workspace/src/ts/workspace/workspace.ts

```diff
@@ -32,21 +32,145 @@ const DEFAULT_WORKSPACE_SIZE = [1, 3];
 
 let ID_COUNTER = 0;
 
-export interface PerspectiveLayout<T> {
-    children?: PerspectiveLayout<T>[];
-    widgets?: T[];
-    sizes: number[];
-}
+export type PerspectiveLayout =
+    | {
+          type: "split-area";
+          sizes: number[];
+          orientation: "horizontal" | "vertical";
+          children: PerspectiveLayout[];
+      }
+    | {
+          type: "tab-area";
+          currentIndex: number;
+          widgets: string[];
+      };
 
 export interface ViewerConfigUpdateExt extends ViewerConfigUpdate {
     table: string;
 }
 
-export interface PerspectiveWorkspaceConfig<T> {
+// TODO: I have made some changes to this interface because it seemed to not reflect reality.
```

Comment:
Lumino's `DockPanel.saveLayout()` does not return string identifiers for `widgets` field, ergo the types for e.g. `workspace.save()` is now not `PerspectiveWorkspaceConfig` (it's an anonymous struct type). Add the explicit returns types to `save()` will show this error.